pt_os_web_portal/app.py

```diff
@@ -39,6 +48,26 @@ def __init__(self, test_mode):
     def start(self):
         self.os_updater.start()
 
+        if (
+            is_pi_top_os()
+            and state.get("app", "onboarded", fallback="false") == "false"
+        ):
+            logger.info("Onboarding not completed ...")
+            if (
+                self.device == DeviceName.pi_top_4.value
+                and state.get("onboarding", "start_miniscreen_app", fallback="false")
+                == "true"
+            ):
+                environ["PT_MINISCREEN_SYSTEM"] = "1"
+                logger.debug("Setting ENV VAR to use miniscreen as system...")
+
+                logger.info("Starting miniscreen onboarding application")
+                self.miniscreen_onboarding = OnboardingAssistantApp()
+                self.miniscreen_onboarding.start()
+            else:
+                logger.info("Not a pi-top[4] - disabling AP mode")
+                disable_ap_mode()
```

Comment:
this should be split out into some other logic - otherwise we are disabling ap on every non-onboarding boot